itest/assets_test.go

```diff
@@ -956,6 +964,16 @@ func defaultPayConfig() *payConfig {
 
 type payOpt func(*payConfig)
 
+func withMaybeGroupKey(groupMode bool, groupKey []byte) payOpt {
+	return func(c *payConfig) {
+		if !groupMode {
+			return
+		}
+
+		c.groupKey = groupKey
+	}
+}
```

Comment:
`c.groupKey` defaults to `nil`, so we could just drop the "Maybe" and `groupMode` arg and just pass in `groupKey == nil` when not in group mode?

Reply:
No problem!

I think you also have the "Maybe" term still in the function name, I think you should remove that because the "maybe" aspect is up to the caller.